.github/workflows/release.yml

```diff
@@ -55,35 +55,54 @@ jobs:
     needs: tag-n-release
     steps:
     - name: Checkout feature-rich code
-      run: |
-        git clone --depth=1 --shallow-submodules --recurse-submodules $GH_REPO -b $REL_TAG --single-branch
-        cd nrn
-        LOCAL_TAG=`git tag`
-        if [ $REL_TAG != $LOCAL_TAG ]; then
-          echo "Wrong tag downloaded!"
-          exit 1
-        else
-          git log --oneline
-        fi
-
-    - name: Make nrnversion.h
-      run: |
-        mkdir build && cd build
-        cmake -DNRN_ENABLE_PYTHON=OFF -DNRN_ENABLE_RX3D=OFF -DNRN_ENABLE_MPI=OFF -DNRN_ENABLE_INTERVIEWS=OFF ../nrn
-        make nrnversion_h VERBOSE=1
-
-    - name: Create nrn-full-src-package
-      id: tar
-      run: |
-        tar -czvf nrn-full-src-package-${REL_TAG}.tar.gz nrn
-        echo "asset_file=nrn-full-src-package-${REL_TAG}.tar.gz" >> $GITHUB_OUTPUT
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+        fetch-depth: 1
 
     - name: Upload nrn-full-src-package to release
       run: |
-        gh release upload ${{ needs.tag-n-release.outputs.rel_tag }} ${{ steps.tar.outputs.asset_file }}
+        filename="nrn-full-src-package-${{ needs.tag-n-release.outputs.rel_tag }}.tar.gz"
+        topdir="nrn-${{ needs.tag-n-release.outputs.rel_tag }}"
+        # taken from:
+        # https://stackoverflow.com/a/67389978
+        git ls-files --recurse-submodules > nrn_files
+        # we also need the .git dir for now
+        find .git >> nrn_files
+        cat nrn_files | tar caf ${filename} --xform s:^:${topdir}/: -T-
+        gh release upload ${{ needs.tag-n-release.outputs.rel_tag }} ${filename}
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  test-full-src-package:
+    runs-on: ubuntu-latest
+    needs: nrn-full-src-package
+    steps:
+      - name: Install apt packages
+        run: |
+          sudo apt-get install -y bison flex libreadline-dev libfl-dev libmpich-dev
+
+      - name: Setup cmake
+        uses: jwlawson/actions-setup-cmake@v2
+        with:
+          cmake-version : '3.18'
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Build nrn-full-src-package
+        run: |
+          gh release download ${{ needs.tag-n-release.outputs.rel_tag }} --pattern 'nrn-full-src-package*' --output nrn.tar.gz
+          tar xf nrn.tar.gz
+          python -m pip install -r nrn/nrn_requirements.txt
```

Comment:
one more detail here and in line 101. Ie. the explicit assumption that the prefix folder is nrn. I don't understand how/why https://github.com/neuronsimulator/nrn/actions/runs/14390297862/job/40355647295 can declare success, given that the extraction of nrn-full-src-package-8.2.7.dev.tar.gz
is into nrn-8.2.7.dev

Reply:
There were a number of  side effect mistakes in setting a prefix folder for tar.gz extraction. A working version branching off this PR is in branch https://github.com/neuronsimulator/nrn/tree/hines/fullsrc